llvm/lib/Target/ARM/MCTargetDesc/ARMMCCodeEmitter.cpp

```diff
@@ -770,7 +770,7 @@ getUnconditionalBranchTargetOpValue(const MCInst &MI, unsigned OpIdx,
   if(MO.isExpr())
     return ::getBranchTargetOpValue(MI, OpIdx, ARM::fixup_t2_uncondbranch, Fixups, STI);
   else 
-    Val = MO.getImm() >> 1;
+    Val = (MO.getImm() - MI.getAddress() - 4) >> 1;
```

Comment:
this function is for both Arm & Thumb. can you confirm this does not break Arm code?

Reply:
Sorry, but I think you're wrong. The only place this function is called is from ARM::t2B instructions. In any case, no new regression tests fail. Have you thought about adding them to the continuous integration?